app.rb

```diff
@@ -0,0 +1,74 @@
+require_relative './lib/item_repository'
+require_relative './lib/order_repository'
+require_relative './lib/database_connection'
+
+class Application
+    database_name = 'store_manager_test'
+
+  def initialize(database_name, io, item_repository, order_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @item_repository = item_repository
+    @order_repository = order_repository
+  end
+
+  def run
+    @io.puts "Welcome to the shop management program! \n\n"
+    @io.puts "What do you want to do?
+      1 = list all shop items
+      2 = create a new item
+      3 = list all orders
+      4 = create a new order"
+     choice = @io.gets.chomp
+    case choice
+    when '1'
+      @item_repository.all.each do |item|
+        @io.puts "##{item.id} #{item.product} - Unit price: #{item.price} - Quantity: #{item.quantity}"
+        run
+      end
+    when '2'
+      item = Item.new
+      @io.puts "Enter name of product"
+      item.product = @io.gets.chomp
+      @io.puts "Enter unit price"
+      item.price = @io.gets.chomp
+      @io.puts "Enter quantitity"
+      item.quantity = @io.gets.chomp
+      @item_repository.add(item)
+      run
+    when '3'
+      @order_repository.all.each do |order|
+        @io.puts "#{order.customer} - #{order.date} - #{order.item_id}"
```

Comment:
good that it shows the id of the item in each order, maybe better to show name of item / both id and name of item